### PR TITLE
Change JSON output format for easy parsing

### LINKED ISSUE #4: Predictable json schema
Hello,

I'm using your tools as mean to extract plugins schemas but since json output depends largely on named attributes and block_types it's harder to parse.

I've reworked the code (can be found here: https://github.com/pchalamet/tfschema) in order to output this json format to be easily machine parseable. Here is an output example for `tfschema resource show -format=json azurerm_virtual_network`

```
{
    "attributes": [
        {
            "name": "address_space",
            "type": "List(String)",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        {
            "name": "dns_servers",
            "type": "List(String)",
            "required": false,
            "optional": true,
            "computed": false,
            "sensitive": false
        },
        ...
    ],
    "block_types": [
        {
            "name": "subnet",
            "attributes": [
                {
                    "name": "address_prefix",
                    "type": "String",
                    "required": true,
                    "optional": false,
                    "computed": false,
                    "sensitive": false
                },
                {
                    "name": "security_group",
                    "type": "String",
                    "required": false,
                    "optional": true,
                    "computed": false,
                    "sensitive": false
                },
                ...
            ],
            "block_types": [],
            "nesting": 3,
            "min_items": 0,
            "max_items": 0
        }
    ]
}
```

This means attributes have no more children of unknown keys - it's an array of well defined structure (new field is `name`). Same for block_types.

Before submitting a PR, I would like to be sure you agree with aforementioned changed. If you agree, I will send you the PR.

Thanks btw for your cool work !

### PR DESCRIPTION
**Pull request by minamijoyo**
Fixes #4

The previous JSON output format depends on the named attributes and block_types. It's hard to parse. If we use a slice instead of a map, we can make the key name predictable. However, I don't want to change the internal data structure for the convenience of the only JSON format. I want to also use a tfschema as a Go library not only CLI. So, the data model of tfschema should be separated from the requirements of the output format.

Extract format related code from the `tfschema` package to a new `formatter` package, define a new type for output and call it from `command` package, and change the JSON output format.
In order to reduce the possibility of future conflicts, the name of keys are borrowed from the schema definition of the grpc provider's proto file.

## before

```
$ tfschema --version
0.1.2

$ tfschema resource show -format=json azurerm_virtual_network
{
    "attributes": {
        "address_space": {
            "type": "List(String)",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        "dns_servers": {
            "type": "List(String)",
            "required": false,
            "optional": true,
            "computed": false,
            "sensitive": false
        },
        "location": {
            "type": "String",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        "name": {
            "type": "String",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        "resource_group_name": {
            "type": "String",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        "tags": {
            "type": "Map(String)",
            "required": false,
            "optional": true,
            "computed": true,
            "sensitive": false
        }
    },
    "block_types": {
        "subnet": {
            "attributes": {
                "address_prefix": {
                    "type": "String",
                    "required": true,
                    "optional": false,
                    "computed": false,
                    "sensitive": false
                },
                "name": {
                    "type": "String",
                    "required": true,
                    "optional": false,
                    "computed": false,
                    "sensitive": false
                },
                "security_group": {
                    "type": "String",
                    "required": false,
                    "optional": true,
                    "computed": false,
                    "sensitive": false
                }
            },
            "block_types": {},
            "nesting": 3,
            "min_items": 0,
            "max_items": 0
        }
    }
}
```

## after

```
$ make build
$ make install
$ $GOPATH/bin/tfschema resource show -format=json azurerm_virtual_network
{
    "attributes": [
        {
            "name": "address_space",
            "type": "List(String)",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        {
            "name": "dns_servers",
            "type": "List(String)",
            "required": false,
            "optional": true,
            "computed": false,
            "sensitive": false
        },
        {
            "name": "location",
            "type": "String",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        {
            "name": "name",
            "type": "String",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        {
            "name": "resource_group_name",
            "type": "String",
            "required": true,
            "optional": false,
            "computed": false,
            "sensitive": false
        },
        {
            "name": "tags",
            "type": "Map(String)",
            "required": false,
            "optional": true,
            "computed": true,
            "sensitive": false
        }
    ],
    "block_types": [
        {
            "type_name": "subnet",
            "attributes": [
                {
                    "name": "address_prefix",
                    "type": "String",
                    "required": true,
                    "optional": false,
                    "computed": false,
                    "sensitive": false
                },
                {
                    "name": "name",
                    "type": "String",
                    "required": true,
                    "optional": false,
                    "computed": false,
                    "sensitive": false
                },
                {
                    "name": "security_group",
                    "type": "String",
                    "required": false,
                    "optional": true,
                    "computed": false,
                    "sensitive": false
                }
            ],
            "block_types": [],
            "nesting": 3,
            "min_items": 0,
            "max_items": 0
        }
    ]
}
```

### CODE REVIEW

**Comment by pchalamet** [comment]
Perfect !

**Comment by minamijoyo** [comment]
@pchalamet Thank you for checking this.

This has been released in v0.2.0 🎉
